C 13 on teaching and learning data science

jrosen48 opened the pull request:
Opening this as I work on chapter 13. This is presently a *very* rough draft.

jrosen48 commented:
Question: do we reference our three pedagogical principles (problem-based learning, differentiation, and universal design for learning) elsewhere? If not, perhaps we can here.

ivelasq commented on 13-students-doing-data-science.Rmd:
```diff
@@ -84,60 +84,63 @@ For example,
+### Consider representation in the data and examples you use
 
-This includes examples, data sets.
+One way to think about data is that it is objective, free of decisions about what to value or prioritize. Another is to consider data as a process that is value-laden, from deciding what question to ask (and what data to collect) to interpreting findings with attention to how others will make sense of them (e.g., O'Neill's *Weapons of Math Destruction*, and Lehrer, Kim, and Schauble's (2007) desc ...
```
I love this! This touches on the equity discussion we had yesterday and we can think of how to thread the needle between all the times it's brought up. I think it's worth expanding this a bit and show examples of how this sort of awareness is needed throughout the entire data science process.

jrosen48 replied:
thanks much - I was proud of this (and also thought, and think, it can be expanded). Examples will help. Will work on this.

jrosen48 replied:
just began to address, but added: "For example, instructors can choose data that directs attention to issues-- equity-related issues in education, for example--that she or he believes would be valuable for students to analyze."

ivelasq commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+- [GAISE Guidelines](https://www.amstat.org/asa/education/Guidelines-for-Assessment-and-Instruction-in-Statistics-Education-Reports.aspx): guidelines for teaching statistics
+- [Teaching Tech Together](http://teachtogether.tech/#the-rules): a book on teaching programming
+- [Data Science for Undergraduates](https://www.nap.edu/catalog/25104/data-science-for-undergraduates-opportunities-and-options): a report on undergraduate data science education
```
Possible addition is the new R Studio education site?

jrosen48 replied:
a *great* idea

jrosen48 replied:
added

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+Thus, the text features walkthroughs that reflect the types of challenges that educational data scientists may encounter in the course of their work: readers may choose to go about approaching the analysis of the data used in each walkthrough differently. At the end of walkthrough, there exist exercises that provide the opportunity for readers to extend and apply the ideas presented in the chapte ...
+
+<!-- To what extent do we need to change the walkthroughs or their order to make this accurate? -->
```
The walk-throughs - at this point in time - all have what I would consider "good core content." That is, they have the central aim of the walk-through written down, but will benefit from editing both in terms of overall arcs of the book as well as adding content that will more accurately reflect this statement.

jrosen48 replied:
adding this as a note to doc to (consistently) return to

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+<!-- To what extent do we need to change the walkthroughs or their order to make this accurate? -->
+
+<!-- Wonder if we add another paragraph about how others can apply problem-based learning in any setting? -->
```
Might be out of scope - but talking more about the general principles of PBL within a data science//programming setting might be useful (I'm thinking mostly about metacognitive skills)

jrosen48 replied:
added as a comment (and added a note in our meeting notes about adding comments for when we begin editing)

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+Differentiation is a method for providing multiple pathways for learners to engage with, understand, and ultimately apply new content and new skills. To differentiate this text, we first created personas for who we expected to be common groups of readers of the book (see [Wilson, 2019](http://teachtogether.tech), for an example). We then differentiate the book by recognizing and providing backgro ...
+
+<!-- Say more about this - specifically how we use the user personas? -->
```
We haven't (yet) added the differentiation components - but I think that this section provides a strong guiding set of principles for editing the text. It's also related to @restrellado suggestion of a heuristic - looking for PBL, Differentiation, and UbD could be added!

jrosen48 replied:
added to notes

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+Part of differentiating this book concerns for whom we are differentiating it. We consider inclusivity (in terms of who belongs as a part of the audience for this text and how this broader view of who participates in data science implies the types of challenges, topics, and data that we include in the book) and accessibility (technically, in terms of how a wide audience of readers is able to acce ...
+
+<!-- Same as above - do we say more about how other teachers can differentiate their instruction? -->
```
Out of scope - although I think we could say more about who we think will benefit from this book (as much as we would love to write a book for _everyone_, there is a more defined persona in our audience(s).)

jrosen48 replied:
know and am happy that we've started working on this!

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+### Universal Design 
+
+Universal design is a series of principles which guide the creation of spaces that are inclusive and accessible to individuals from all walks of life regardless of age, size, ability, or disability. While traditionally applied to physical spaces, we have extended these principles to the creation of this book in such a way that the text and accompanying materials will be designed for individuals f ...
```
we're going to need to think through this a bit more, because I don't know that we have the resources in place to do this in terms of language, screen readers, educational background, etc. I love this as an aspiration, and I think we can make progress towards this, but it will likely be an ongoing process rather than something that is ready at the time of publication.

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+Many of the seven guiding principles of Universal Design are readily transferable to the creation of a text, such as equitable use, flexibility in use (aided in large part through differentiation), simple and intuitive use, perceptible information, and tolerance for error.
+
+<!-- Say more regarding how we apply these? -->
```
more for our heuristics in editing!

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+We think this consideration is important - particularly when it comes to data-related issues that we consider to be objective, such as how variables are assumed and constructed to be dichotomous (such as variables for individual's gender) or categorical (such as variables for individual's race), when the truth may be that such variables are based on decisions that analysts or those collecting dat ...
+
+### Provide a home base for learners to access resources (and to learn more)
```
this is fantastic - and maybe should lead the section? we should also be sure that we're providing a "home base" in our text.

jrosen48 replied:
move to start. great/cool idea about a home base in our text, adding to notes

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+## General strategies related to teaching and learning
+
+The National Academy of Science commissioned a report, [*How People Learn*](https://www.nap.edu/catalog/9853/how-people-learn-brain-mind-experience-and-school-expanded-edition), that aimed to summarize research from the educational psychology and the learning sciences on teaching and learning. In 2018, the book was updated in [*How People Learn II*](https://www.nap.edu/catalog/24783/how-people-le ...
```
what I'm not seeing in our resources are pedagogical texts related to classroom instruction - I would argue that K-12 education has largely provided the answers to "How do we teach data science?" question, but yet data science continues to recreate the wheel. 

I would add in resources by Hattie (Visible Learning), Lemov (TLAC 2.0), and potentially Bambrick, along with "Understanding by Design Meets Neuroscience" by McTighe and Willis (the original, Understanding by Design, was done by Wiggins & McTighe)

jrosen48 replied:
helpful. added as comments and will discuss in meeting so as not to let this slide to editing.

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+We often think of learning in terms of objectives for specific lessons, but learners learn many different things at different times. The authors of *HPL2* point out that individuals learn in response to different challenges and circumstances, including those in formal learning environements, such as workshops or classes. This principle implies a strategy that involves supporting learners to do da ...
+
+### Metacognition is important (even though it sounds more sophisticated than it is!)
```
I'm wondering how we can model metacognition throughout the text using callouts, etc.

jrosen48 replied:
added to notes

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+Educators know that how motivated learners are matters. According to the authors of *HPL2*, teachers can make an impact on how motivated learners are. Some specific things that educators can do to support learners include helping learners to set (and to work toward) goals, selecting content that is valuable and interesting to learners, helping learners to have choices and showing them how they ar ...
+
+## Summary
```
overall I think this is a fantastic start to this topic. my big wondering is how we can make it more actionable, as it reads largely as theoretical. I'm wondering if, based on how we're setting up the text, it would be more effective to provide context (as in the various paragraphs) and then 2 - 5 action steps, so that readers can choose what works for them and immediately start to put it into practice.

jrosen48 replied:
great call. added as comments and notes.

kierisi commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+In this section, we described the pedagogical principles for this book and strategies for teaching data science. We also directed attention to more general strategies for teaching and learning. Teaching data science is a relatively new area, but data science educators are not alone, given resources that are beign developed and those that can be adapted from other disciplines and the wider body of ...
+
+<!-- Should we include notes on teaching K-12 students? -->
```
that could be its own book!

jrosen48 commented on 13-students-doing-data-science.Rmd:
```diff
@@ -1 +1,146 @@
+
+### Problem-based learning 
+
+Problem-based learning (PBL) is a method of instruction that presents learners with a real-world challenge in which they must apply their skills and knowledge to solve. We applied this principle to the design of this book through including walk-throughs for common questions with respect to data science and education. Whereas some topics may benefit less from such an approach (and the inclusion of ...
```
added to notes to discuss

jrosen48 commented:
Thanks again @ivelasq and @kierisi for great feedback. I a) addressed feedback when possible and b) added notes in markdown and notes in our meeting doc to discuss how to address those I couldn't right away.
